File: 217.contains-duplicate.py
#
# @lc app=leetcode id=217 lang=python3
#
# [217] Contains Duplicate
#

# @lc code=start
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        dic = set()

        for num in nums:
            if num in dic:
                return True
            dic.add(num)
            
        return False
# @lc code=end

File: 217.contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> sett = new HashSet<>();

        for (int num : nums) {
            if (sett.contains(num)) {
                return true;
            }
            sett.add(num);
        }
        return false;
    }
}